src/analysis_and_optimization/Partial_evaluator.ml

```diff
@@ -22,7 +22,7 @@ let apply_prefix_operator_int (op : string) i =
         | "PPlus__" -> i
         | "PMinus__" -> -i
         | "PNot__" -> if i = 0 then 1 else 0
-        | s -> raise_s [%sexp (s : string)] ) )
+        | s -> Common.FatalError.fatal_error_msg [%message s] ) )
```

Comment:
A stupid question from someone who knows nothing about Sexplib: what's the difference between `[%sexp ...]` and `[%message ...]`? `Optimize.ml` has one fatal error that keeps `[%sexp ...]` but here in `Partial_evaluator.ml` they're all changed to `[%message ...]`.

Reply:
The difference is subtle and honestly seems kind of arbitrary. `[%message` comes from https://github.com/janestreet/ppx_sexp_message, and `[%sexp` comes from https://github.com/janestreet/ppx_sexp_value

They have slightly different behavior for how they label their values, which is relevant for when the payload is only a string. `[%sexp (s:string)]` will output something like `(s ("The contents of the string"))`, while `[%message s]` should just output `The contents of the string`, but they accomplish the same goal in the end